x/gov/keeper/deposit.go

```diff
@@ -203,6 +203,8 @@ func (keeper Keeper) ChargeDeposit(ctx sdk.Context, proposalID uint64, destAddre
 				return err
 			}
 		}
+
+		store.Delete(types.DepositKey(deposit.ProposalId, depositerAddress))
```

Comment:
Say a function below fails (`BurnCoins`, `GetModuleAddress`, `FunCommunityPool`), what happens to the deposit then?

Isn't it safer to store the deposit keys to delete and delete them at the end of his function?

Reply:
I think we might not need to do that, because if we return an error all the changes done during the message execution won't get persisted, right? But now that you mention it I'm not that sure